contracts/BC_fusion/AirDrop.sol

```diff
@@ -0,0 +1,146 @@
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+import "./interface/ITokenHub.sol";
+import "./interface/IAirDrop.sol";
+import "./System.sol";
+import "./lib/Utils.sol";
+
+contract AirDrop is IAirDrop, ReentrancyGuardUpgradeable, System {
+    string public constant sourceChainID = "Binance-Chain-Ganges";
+    address public approvalAddress = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa;
+    bytes32 public override merkleRoot = 0x0000000000000000000000000000000000000000000000000000000000000000;
+    bool public merkleRootAlreadyInit = false;
+
+    // claimedMap is used to record the claimed token.
+    mapping(bytes32 => bool) private claimedMap;
+
+    function isClaimed(bytes32 node) public view override returns (bool) {
+        return claimedMap[node];
+    }
+
+    function claim(
+        bytes32 tokenSymbol, uint256 amount,
+        bytes calldata ownerPubKey, bytes calldata ownerSignature, bytes calldata approvalSignature,
+        bytes32[] calldata merkleProof) nonReentrant external override {
+        // Recover the owner address and check signature.
+        bytes memory ownerAddr = _verifyTMSignature(ownerPubKey, ownerSignature, _tmSignatureHash(tokenSymbol, amount, msg.sender));
+        // Generate the leaf node of merkle tree.
+        bytes32 node = keccak256(abi.encodePacked(ownerAddr, tokenSymbol, amount));
+    
+        // Check if the token is claimed.
+        require(isClaimed(node), "AlreadyClaimed");
+
+        address contractAddr = address(0x00);
+        // Check if the token is exist.
+        if (tokenSymbol != ITokenHub(TOKEN_HUB_ADDR).BEP2_TOKEN_SYMBOL_FOR_BNB()) {
```

Comment:
I dont think we need handle this in this contract, let `ITokenHub(TOKEN_HUB_ADDR).unlock` do this inside the tokenhub contract. We just pass the tokenSymbol.

Reply:
fixed